src/charm.py

```diff
@@ -18,3 +24,56 @@ def __init__(self, *args):
             args: Arguments passed to the CharmBase parent constructor.
         """
         super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.saml_relation_created, self._on_relation_created)
```

Comment:
Should the function name match the event name?

Reply:
Fixed. Thx